pymc_experimental/model/marginal_model.py

```diff
@@ -208,35 +210,50 @@ def logp(self, vars=None, **kwargs):
             vars = [m[var.name] for var in vars]
         return m._logp(vars=vars, **kwargs)
 
-    def clone(self):
-        m = MarginalModel(coords=self.coords)
-        model_vars = self.basic_RVs + self.potentials + self.deterministics + self.marginalized_rvs
-        data_vars = [var for name, var in self.named_vars.items() if var not in model_vars]
+    @staticmethod
+    def _clone(model: Union[Model, "MarginalModel"]):
+        new_model = MarginalModel(coords=model.coords)
+        if isinstance(model, MarginalModel):
+            marginalized_rvs = model.marginalized_rvs
+            marginalized_named_vars_to_dims = model._marginalized_named_vars_to_dims
+        else:
+            marginalized_rvs = []
+            marginalized_named_vars_to_dims = {}
+
+        model_vars = model.basic_RVs + model.potentials + model.deterministics + marginalized_rvs
+        data_vars = [var for name, var in model.named_vars.items() if var not in model_vars]
         vars = model_vars + data_vars
         cloned_vars = clone_replace(vars)
         vars_to_clone = {var: cloned_var for var, cloned_var in zip(vars, cloned_vars)}
-        m.vars_to_clone = vars_to_clone
-
-        m.named_vars = treedict({name: vars_to_clone[var] for name, var in self.named_vars.items()})
-        m.named_vars_to_dims = self.named_vars_to_dims
-        m.values_to_rvs = {i: vars_to_clone[rv] for i, rv in self.values_to_rvs.items()}
-        m.rvs_to_values = {vars_to_clone[rv]: i for rv, i in self.rvs_to_values.items()}
-        m.rvs_to_transforms = {vars_to_clone[rv]: i for rv, i in self.rvs_to_transforms.items()}
-        m.rvs_to_initial_values = {
-            vars_to_clone[rv]: i for rv, i in self.rvs_to_initial_values.items()
+        new_model.vars_to_clone = vars_to_clone
+
```

Comment:
Should we be copying over coords as well? This might address #315

Reply:
We are already. Above `coords=model.coords`, and the mutable dim_lengths are part of `data_vars`